messaging/src/main/scala/uk/ac/wellcome/messaging/sns/NotificationMessage.scala

```diff
@@ -3,5 +3,10 @@ package uk.ac.wellcome.messaging.sns
 import io.circe.generic.extras.JsonKey
 
 case class NotificationMessage(
-  @JsonKey("Message") body: String
+  @JsonKey("Message") body: String,
+  MessageAttributes: Option[Map[String, AttributeValue]] = None
+)
```

Comment:
Whilst we are working on this, is there any reason we can't change this to be:

```scala
case class NotificationMessage[T](
  body: T,
 ...
)
```

There are 2 really annoying things about the current implementation:
1) We have nested json encoded strings within `body`, with illegible amounts of escaping of quotes etc
2) We always end up decoding `T` in the application code, whereas this could be handled by the library instead

Of course if we did do this, we would need to have a `Decoder[T]` or `Encoder[T]` specified when using `NotifactionMessage`, but in practice we always have these defined.

Reply:
Yeh I know. There are quite a lot of things around sending messages and retrieving messages that are confusing. No objections to changing it but I'm trying to limit the changes in this PR because it's big and also because I want to get this done. Any objections to doing this elsewhere?

Reply:
Nope thats fine